palette/reverse.go

```diff
@@ -0,0 +1,30 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package palette
+
+import (
+	"image/color"
+)
+
+// Reverse is a ColorMap that reverses the direction of the ColorMap it
+// contains.
+type Reverse struct {
```

Comment:
Instead of exposing a type, perhaps do like the stdlib's 'sort.Reverse' that exposes a function and returns an unexposed type that implements the same interface?